### PR TITLE
refactor: refactor replica head and snap chain related implementation

### PR DESCRIPTION
**Pull request by shuo-wu**
#### What this PR does / why we need it:

Considering there may be broken replicas reusage, w need a better way to handle replica head and snap chain

### CODE REVIEW

**Review by DamiaSan — DISMISSED** [review]
LGTM

**Comment by derekbit** [comment]
LGTM @shuo-wu ready to merge?

**Comment by shuo-wu** [comment]
@derekbit YES it's ready to go

**Comment by shuo-wu** [comment]
@derekbit @c3y1huang The unit test and rebuilding e2e test are passed. Plz take a look

File: pkg/spdk/replica.go [hunk 1]
```diff
@@ -41,13 +41,13 @@ type Replica struct {
 
 	ctx context.Context
 
+	// Head should be the only writable lvol in the regular Replica lvol chain/map.
+	// And it is the last entry of ActiveChain if it is not nil.
+	Head *Lvol
 	// ActiveChain stores the backing image info in index 0.
 	// If a replica does not contain a backing image, the first entry will be nil.
-	// The last entry of the chain is always the head lvol.
+	// The last entry of the chain should the head lvol if it exists.
```
Comment (ChanYiLin) [inline pkg/spdk/replica.go:49]: typo: should "be"

**Review by derekbit — APPROVED** [review]

**Pull request merged**